contracts/RiskSteward/RiskStewardReceiver.sol

```diff
@@ -0,0 +1,322 @@
+import { IRiskOracle, RiskParameterUpdate } from "../interfaces/IRiskOracle.sol";
+import { IVToken } from "../interfaces/IVToken.sol";
+import { ICorePoolComptroller } from "../interfaces/ICorePoolComptroller.sol";
+import { IIsolatedPoolsComptroller } from "../interfaces/IIsolatedPoolsComptroller.sol";
+import { IRiskStewardReceiver, RiskParamConfig } from "./IRiskStewardReceiver.sol";
+
+/**
+ * @title RiskStewardReceiver
+ * @author Venus
+ * @notice Contract that can automatically adjust market caps based on risk oracle recommendations
+ * @custom:security-contact https://github.com/VenusProtocol/governance-contracts#discussion
+ */
+contract RiskStewardReceiver is IRiskStewardReceiver, Initializable, Ownable2StepUpgradeable {
+    /**
+     * @notice Mapping of supported risk configurations and their validation parameters
+     */
+    mapping(string updateType => RiskParamConfig) private riskParameterConfigs;
+
+    /**
+     * @notice Whitelisted oracle address to receive updates from
+     */
+    IRiskOracle public immutable RISK_ORACLE;
+
+    /**
+     * @notice Address of the CorePool comptroller used for selecting the correct comptroller abi
+     */
+    ICorePoolComptroller public immutable CORE_POOL_COMPTROLLER;
+
+    /**
+     * @notice Mapping of market and update type to last update. Used for debouncing updates.
+     */
+    mapping(bytes marketUpdateType => uint256) public lastProcessedTime;
+
+    /**
+     * @notice Mapping of processed updates. Used to prevent re-execution
+     */
+    mapping(uint256 updateId => bool) public processedUpdates;
+
+    /**
+     * @notice Time before a submitted update is considered stale
+     */
+    uint256 public constant UPDATE_EXPIRATION_TIME = 1 days;
+
+    /**
+     * @notice Event emitted when an update is successfully applied
+     */
+    event RiskParameterUpdated(uint256 updateId);
+
+    /**
+     * @notice Flag for if a risk parameter type can be updated by the steward
+     */
+    error ConfigNotActive();
+
+    /**
+     * @notice Error for when an update was not applied within the required timeframe
+     */
+    error UpdateIsExpired();
+
+    /**
+     * @notice Thrown when an update has already been processed
+     */
+    error ConfigAlreadyProcessed();
+
+    /**
+     * @notice Thrown when the debounce period hasn't passed for applying an update to a specific market/ update type
+     */
+    error UpdateTooFrequent();
+
+    /**
+     * @notice Thrown when the new value of an update is our of range
+     */
+    error UpdateNotInRange();
+
+    /**
+     * @notice Thrown when a config that is not implemented is set
+     */
+    error UnsupportedUpdateType();
+
+    /**
+     * @notice Flag to pause/unpause update processing
+     */
+    bool public paused;
+
+    /**
+     * @notice Event emitted when pause state is changed
+     */
+    event PauseStateChanged(bool newState);
```

Comment:
Would it make sense to use `@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol` instead?